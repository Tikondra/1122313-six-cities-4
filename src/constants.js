export const PLACES = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
];

export const Settings = {
  COUNT_PLACES: 512,
};
