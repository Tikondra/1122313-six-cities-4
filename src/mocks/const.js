export const Cities = [
  {
    title: `Paris`,
    isActive: true,
  },
  {
    title: `Cologne`,
    isActive: false,
  },
  {
    title: `Brussels`,
    isActive: false,
  },
  {
    title: `Amsterdam`,
    isActive: false,
  },
  {
    title: `Hamburg`,
    isActive: false,
  },
  {
    title: `Dusseldorf`,
    isActive: false,
  },
];

export const Settings = {
  COUNT_PLACES: 512,
};
