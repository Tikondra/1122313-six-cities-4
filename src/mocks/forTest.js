import {TypePlace} from "../constants";

export const offers = [
  {
    id: `bkjnlkmlkms`,
    coordinates: [48.853438, 2.349984],
    city: `Paris`,
    isPremium: true,
    price: 35,
    title: `Beautiful & luxurious apartment at great location`,
    type: TypePlace.APARTMENT,
    rating: 1.3,
    picture: `img/apartment-02.jpg`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-01.jpg`, `img/apartment-03.jpg`],
    description: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    room: 3,
    guests: 3,
    options: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Clara`,
      isSuper: false,
    },
    reviews: [
      {
        author: `Wilfred`,
        avatar: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        date: new Date(2020, 2, 15),
      },
      {
        author: `Alfred`,
        avatar: `img/avatar-max.jpg`,
        rating: 2,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        date: new Date(2020, 5, 23),
      },
      {
        author: `Tom`,
        avatar: `img/avatar-max.jpg`,
        rating: 2,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        date: new Date(2020, 3, 23),
      },
    ]
  },
  {
    id: `jkbsdkjsjldk`,
    coordinates: [52.3809553943508, 4.939309666406198],
    city: `Amsterdam`,
    isPremium: false,
    price: 25,
    title: `Wood and stone place`,
    type: TypePlace.HOTEL,
    rating: 2.5,
    picture: `img/apartment-01.jpg`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-01.jpg`, `img/apartment-03.jpg`],
    description: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    room: 3,
    guests: 3,
    options: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Clara`,
      isSuper: false,
    },
    reviews: [
      {
        author: `Jon`,
        avatar: `img/avatar-max.jpg`,
        rating: 5,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        date: new Date(2020, 3, 25),
      },
      {
        author: `Rory`,
        avatar: `img/avatar-max.jpg`,
        rating: 3,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        date: new Date(2020, 4, 5),
      },
    ]
  },
];

export const cities = [
  {
    title: `Paris`,
    isActive: false,
    coordinates: [48.856663, 2.351556]
  },
  {
    title: `Cologne`,
    isActive: false,
    coordinates: [47.010209, 2.320878]
  },
  {
    title: `Brussels`,
    isActive: false,
    coordinates: [50.851309, 4.351718]
  },
  {
    title: `Amsterdam`,
    isActive: false,
    coordinates: [52.38333, 4.9]
  },
  {
    title: `Hamburg`,
    isActive: false,
    coordinates: [53.552645, 9.966287]
  },
  {
    title: `Dusseldorf`,
    isActive: false,
    coordinates: [51.230569, 6.787428]
  },
];

export const SortType = {
  POPULAR: `Popular`,
  PRICE_UP: `Price: low to high`,
  PRICE_DOWN: `Price: high to low`,
  TOP_RATING: `Top rated first`
};

export const SORT_TYPES = Object.values(SortType);
