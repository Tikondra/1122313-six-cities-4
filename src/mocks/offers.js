import {nanoid} from "nanoid";
import {TypePlace} from "../constants";

export default [
  {
    id: nanoid(),
    coordinates: [52.3909553943508, 4.85309666406198],
    isPremium: true,
    price: 35,
    title: `Beautiful & luxurious apartment at great location`,
    type: TypePlace.APARTMENT,
    rating: 1.3,
    picture: `img/apartment-02.jpg`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-01.jpg`, `img/apartment-03.jpg`],
    description: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    room: 3,
    guests: 3,
    options: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Clara`,
      isSuper: false,
    }
  },
  {
    id: nanoid(),
    coordinates: [52.369553943508, 4.85309666406198],
    isPremium: false,
    price: 25,
    title: `Wood and stone place`,
    type: TypePlace.HOTEL,
    rating: 2.5,
    picture: `img/apartment-01.jpg`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-01.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    description: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    room: 1,
    guests: 2,
    options: [`Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Donna`,
      isSuper: false,
    }
  },
  {
    id: nanoid(),
    coordinates: [52.3909553943508, 4.929309666406198],
    isPremium: false,
    price: 30,
    title: `Nice, cozy, warm big bed apartment`,
    type: TypePlace.HOUSE,
    rating: 3.7,
    picture: `img/apartment-03.jpg`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-01.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    description: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    room: 2,
    guests: 4,
    options: [`Towels`, `Baby seat`, `Cabel TV`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Marta`,
      isSuper: false,
    }
  },
  {
    id: nanoid(),
    coordinates: [52.3809553943508, 4.939309666406198],
    isPremium: true,
    price: 10,
    title: `Canal View Prinsengracht`,
    type: TypePlace.ROOM,
    rating: 4.1,
    picture: `img/room.jpg`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-01.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`, `img/apartment-01.jpg`],
    description: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    room: 4,
    guests: 1,
    options: [`Washing machine`, `Coffee machine`, `Dishwasher`, `Wi-Fi`, `Heating`, `Kitchen`, `Fridge`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Rose`,
      isSuper: true,
    }
  },
];
