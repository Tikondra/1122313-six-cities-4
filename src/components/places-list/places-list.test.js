import React from "react";
import renderer from "react-test-renderer";
import PlacesList from "./places-list";
import {offers} from "../../mocks/forTest";
import {PlacesListClass} from "../../constants";

it(`should render PlacesList`, function () {
  const tree = renderer
    .create(
        <PlacesList
          className = {PlacesListClass.MAIN}
          offers={offers}
          onClickByHeader={jest.fn()}
          onHoverPlace={jest.fn()}/>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
