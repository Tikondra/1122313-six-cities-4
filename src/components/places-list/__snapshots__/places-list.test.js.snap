// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PlacesList 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <article
    className="cities__place-card place-card"
    id="bkjnlkmlkms"
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Beautiful & luxurious apartment at great location"
          className="place-card__image"
          height="200"
          src="img/apartment-02.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            35
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "20%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[MockFunction]}
      >
        <a
          href="#"
        >
          Beautiful & luxurious apartment at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    id="jkbsdkjsjldk"
    onMouseOver={[Function]}
  >
    
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Wood and stone place"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            25
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[MockFunction]}
      >
        <a
          href="#"
        >
          Wood and stone place
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Hotel
      </p>
    </div>
  </article>
</div>
`;
